src/ipc/ipc-common.c

```diff
@@ -188,13 +188,16 @@ void ipc_msg_send(struct ipc_msg *msg, void *data, bool high_priority)
 		assert(!ret);
 	}
 
+#ifndef __ZEPHYR__
```

Comment:
Difficult to understand the code, it may be easier to follow if we 
```c
if (cpu_is_me() != PRIMARY) {
    // some error log and return value
}
```
This way we will catch anyone that attempts this today and it could be extended in the future to allow more IPC capable cores if needed.

Reply:
@lgirdwood I don't think this code does anything depending on the core, where it's running. It just makes sure, that *no* IPC ever gets sent immediately from here but always from the queue. It is the IPC queue that is always handled by the primary core, whereas submitting to the queue is allowed from any core.

Reply:
@lgirdwood My intention was to queue all IPC's to Host, regardless if this is a primary or secondary core. Even in context of single core, notifications and IPC responses are asynchronous: EDF thread may want send IPC response and be interrupted by LL thread which want to send notification. For single core, current synchronization mechanisms can secure this flow, but for multicore the situation gets more complicated. That is why I think its more consistent and secure if all IPC's are queued on queue executed only on primary core.
Normally I would just remove the part of code trying to send IPC immediately, but there is still XTOS configuration where I think sending IPC's by secondary core was allowed. I did not want to change flow of IPC3/XTOS, as I am focusing on IPC4/Zephyr.

Reply:
@pblaszko Please add an inline comment why you ifdef the code out on Zephyr. Code is pretty hard to follow otherwise.

Reply:
> @lgirdwood I don't think this code does anything depending on the core, where it's running. It just makes sure, that _no_ IPC ever gets sent immediately from here but always from the queue. It is the IPC queue that is always handled by the primary core, whereas submitting to the queue is allowed from any core.

@lyakh Your comment catch the essence of the change, so if you don't mind, I put it to inline comment :)

Reply:
@pblaszko sure, feel free